# debian_alsaconf_ignoreinstall.patch by Thomas Hood
#
# Use modprobe --ignore-install when modprobing legacy card drivers with args
Index: alsa-utils-1.0.18/alsaconf/alsaconf.in
===================================================================
--- alsa-utils-1.0.18.orig/alsaconf/alsaconf.in	2008-11-02 20:59:26.000000000 +0100
+++ alsa-utils-1.0.18/alsaconf/alsaconf.in	2008-11-02 20:59:26.000000000 +0100
@@ -1034,7 +1034,11 @@
 # return 0 - OK, 1 - NG, 2 - not working (irq/dma problem)
 ac_try_load () {
     test -n "$LOGFILE" && echo "$1 ${*:2}" >> "$LOGFILE"
+if [ "$kernel" = old ] ; then
     /sbin/modprobe snd-$1 ${*:2} >/dev/null 2>&1
+else
+    /sbin/modprobe --ignore-install snd-$1 ${*:2} >/dev/null 2>&1
+fi
     if $lsmod | grep -q -E '^(snd-|snd_)'$1' '; then
 	: ;
     else
@@ -1073,7 +1077,11 @@
 # ac_try_capture card duplex opts
 ac_try_capture () {
     test -n "$LOGFILE" && echo "$1 ${*:2}" >> "$LOGFILE"
+if [ "$kernel" = old ] ; then
     /sbin/modprobe snd-$1 ${*:3} >/dev/null 2>&1
+else
+    /sbin/modprobe --ignore-install snd-$1 ${*:3} >/dev/null 2>&1
+fi
     if $lsmod | grep -q -E '^(snd-|snd_)'$1' '; then
 	: ;
     else
